### ISSUE TITLE
Max volume in Kodi with DAC+AMP: Kodi-volume <= Volumio-volume 

### ISSUE THREAD

**Issue by emanuele** [Saiyato/volumio-kodi-plugin#22]
First of all, thanks a lot for this excellent plugin!

With the last update of Volumio, I could start Kodi using the IQAudio DAC+AMP on RPi2, through the plugin. What I observe is that the maximum volume which can be set within Kodi is upper-bound by the volume set in Volumio, from the web interface. I don't see a way to set the volume from Kodi, e.g. via Kodi remote, that overcomes the amount set in Volumio. Is that possible?
In essence: Kodi-volume <= Volumio-volume

Usually, when listening to music, I keep the Volumio's volume to a moderate value, like 20-25. Unfortunately, when using Kodi from Volumio, movies have way too low volume, even when increasing to the maximum level allowed within Kodi. So I start to go back and forth from the Kodi remote to the Volumio web interface, in order to set a max Volumio's volume which is good for the movie/Kodi too (usually around 70-80, depending on the movie), and then tune the movie volume within Kodi while playing the movie. The whole procedure is a bit fiddly and annoying, since it has to be done every time I switch from music to movie. Moreover, if I forget to re-set the Volumio volume to 20-25 when going back to music, I risk to have an super-exceedingly high volume when playing music.

Is it possible to set the general volume of Volumio from within Kodi, when using a DAC+AMP card? 

Notice that when using HDMI audio in Kodi and DAC+AMP in Volumio, the problem does not exist anymore, because the two volume's settings are independent.

**Comment by nicosea** [comment]
Be carefully in doing that. The volume in KODI should always be 100% and you should control the volume during movies by Volumio GUI, so in this way you are using hardware volume control and not software. If you use KODI volume (also by CEC remote TV control), than you will change the volume by software and the audio quality will be lower.

**Comment by emanuele** [comment]
Fair point: hardware volume control is indeed superior to the software one, in many aspects, especially quality. The only downside of setting once for all the volume in Kodi at 100%, is to keep jumping back and forth from one app (Volumio, for volume control) to another (Kodi remote / Yatse, for playback control) when watching a movie. The best would be to use hardware volume control in Kodi. Do you think it is possible?

**Comment by nicosea** [comment]
Ciao Emanuele, I am a user and not a developer, but I agree with you, it would be nice to have a "single" volume hardware control. 

By the way: neither Libreelec or OSMC support hardware volume control if you use them with raspberry PI and DAC (a iqaudio DAC+ in my case). At least with this plugin in Volumio I can use the volumio GUI. With Libreelec I used to send SSH commands... or use the alsamixer "graphical" tool by terminal, to be sure to change hardware volume (in OSMC not even that, since no alsa tools).

**Comment by biva** [comment]
Hi! Any news on this issue? (not a priority for me though, I really like your plugin :))) )

**Comment by emanuele** [comment]
Hi,

I guess there is no solution for this one. Now I'm following the advice of
keeping the Kodi volume to 100% and then just use the Volumio webapp to
tweak the general volume.

BTW, after the update of early January I couldn't set any options of the
Kodi plugin of Volumio, due to a bug, so no DAC+AMP support anymore. I
reported that on the Volumio forum, in the bug section, but no answer since
then. I should have reported here too? I might have been solved in the
meanwhile. News?

Best,

Emanuele

**Comment by Saiyato** [comment]
I'm not sure what you are experiencing, can you guide me through this? What steps did you take to configure the amp?

**Comment by emanuele** [comment]
Hi,

sorry for the late reply. Before last December, on RPi2/3, you could use
the DAC/AMP hat as audio output both for Volumio and Kodi, instead of HDMI
or the audio jack. The volumio-kodi-plugin allowed that choice in the
settings. Unfortunately, after upgrading Volumio, It is not
possibile anymore, see issue #28.

Best,

Emanuele
